packages/roosterjs-editor-api/lib/format/setAlignment.ts

```diff
@@ -3,34 +3,127 @@ import {
     Alignment,
     ChangeSource,
     DocumentCommand,
+    ExperimentalFeatures,
     IEditor,
     QueryScope,
 } from 'roosterjs-editor-types';
 
+const TABLE = 'TABLE';
+const TEXT = 'TEXT';
+
 /**
  * Set content alignment
  * @param editor The editor instance
  * @param alignment The alignment option:
  * Alignment.Center, Alignment.Left, Alignment.Right
  */
 export default function setAlignment(editor: IEditor, alignment: Alignment) {
-    let command = DocumentCommand.JustifyLeft;
-    let align = 'left';
+    const element = editor.getElementAtCursor();
+    const elementType = isListOrATableOrText(element);
+
+    editor.addUndoSnapshot(() => {
+        alignElement(editor, element, elementType, alignment);
+        editor.queryElements('[align]', QueryScope.OnSelection, node =>
+            alignElement(editor, node, elementType, alignment, true /** addUndoSnapshot */)
+        );
+    }, ChangeSource.Format);
+}
+
+/**
+ * Check if the element at the cursor is a table or text element
+ * @param element
+ * @returns
+ */
+function isListOrATableOrText(element: HTMLElement) {
+    if (!element) {
+        return;
+    }
+    const tag = element.tagName;
+    if (tag === 'TABLE') {
+        return TABLE;
+    } else {
+        return TEXT;
+    }
+}
+
+/**
+ * Align the element left, center and right
+ * @param editor
+ * @param element the element being aligned
+ * @param elementType type text or table
+ * @param alignment the alignment type
+ * @param addUndoSnapshot check if this function is being called by addUndoSnapshot
+ */
+function alignElement(
+    editor: IEditor,
+    element: HTMLElement,
+    elementType: string,
+    alignment: Alignment,
+    addUndoSnapshot?: boolean
+) {
+    if (elementType === TABLE && editor.isFeatureEnabled(ExperimentalFeatures.TableAlignment)) {

```

Comment:
We can check whether the whole table is selected using the editor api getSelectionRangeEx().

```
const selection = editor.getSelectionRangeEx();
if (selection.type == SelectionRangeTypes.TableSelection) {
 const vTable = new VTable(selection.table)
 const { firstCell, lastCell } = selection.coordinates;
    const rowsLength = vTable.cells.length - 1;
    const colIndex = vTable.cells[rowsLength].length - 1;
```

    const firstX = firstCell.x;
    const firstY = firstCell.y;
    const lastX = lastCell.x;
    const lastY = lastCell.y;

    const selectedAllTable = firstX == 0 && firstY == 0 && lastX == colIndex && lastY == rowsLength;
```
 if(selectedAllTable) { alignAllTable() }
}
```

Reply:
I see, if I select only some cells, the editor.getElementAtCursor returns Table. I will fix that.